sdk/core/azure-core/tests/test_streaming.py

```diff
@@ -111,6 +113,19 @@ def test_decompress_plain_header(http_request):
     except DecodeError:
         pass
 
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+def test_decompress_plain_header_offline(port, http_request):
+    request = http_request(method="GET", url="http://localhost:{}/streams/compressed".format(port))
+    with RequestsTransport() as sender:
+        response = sender.send(request, stream=True)
+        response.raise_for_status()
+        try:
+            result = response.read()
```

Comment:
Like in the above online test, I think we want to stream download the response similar to L109 with the `decompress=True` keyword argument passed. This should signal that we want to decompress the body based on the Content-Encoding header and we should get the expected behavior of failure

Reply:
with pytest.raises(DecodeError):
   list(data)

Reply:
Included these changes in the most recent commit!